### ISSUE TITLE
Comp is not a function error

### ISSUE THREAD

**Issue by jan-vdm** [solidjs/solid#764]
Whenever I run the dev command it complains about Comp not being a function, however when I build and then serve the project it works perfectly fine with 0 console logs to indicate otherwise. At first I thought it is a component in my logged in state so I cleared my cache and forced a log out but it also happens on the log in screen. This is frustrating as I can still make changes, build a production version and serve it to check but would like to have hot reload working....
<img width="1724" alt="Screen Shot 2021-12-16 at 2 14 17 pm" src="https://user-images.githubusercontent.com/30859194/146302162-e8bc9bb4-2809-44bb-89ae-b1a8b98642be.png">

I checked the auth page and the container context but to be honest this is different error logs popping up vs the logged in state. And about a month ago it was working perfectly fine, just all of a sudden stopped...

Some context, I use contexts and use the children props within my context providers, this hasn't been an issue before but this is practically the only complex thing I've done with solidjs...

**Comment by ryansolid** [comment]
Yeah this mostly like has to do with a bug with the the heuristic the Babel plugin for HMR has. We need to transform components to make HMR work and it is possibly incorrectly identifying a Component or something like that. We've been doing a lot of work here recently. Have you tried updating to the latest `vite-plugin-solid` (and `solid-refresh`). It's also possible this is caused by the latest version of `solid-refresh`. I'd try the latest first and if that doesn't work we need to see if we can narrow it down.

**Comment by jan-vdm** [comment]
@ryansolid I've tried ```"vite-plugin-solid": "2.1.4"``` upgrading from ```2.1.2``` and it doesn't work however, since you pointed out it was this package I rolled back my version to ```2.0.0``` and that seems to work. Thanks for getting back to me quickly

**Comment by LXSMNSYC** [comment]
It narrows down to how you name functions. If you have a function that has Pascal-case format, it is assumed to be a component.
